analyzere/extras/visualizations.py

```diff
@@ -0,0 +1,272 @@
+            if node_hash not in self._unique_nodes:
+                self._unique_nodes[node_hash] = self._sequence()
+
+            terms = self._format_layer_terms(l.sink) if include_terms else ''
+            name = ('Nested {} {}'.format(
+                    l.sink.type, self._format_description(l.description))
+                    if l.description else
+                    'Nested {} {}'.format(
+                    l.sink.type, self._format_description(l.sink.description))
+                    if l.sink.description else
+                    'Nested {} ({}) {}'
+                    .format(l.sink.type, self._unique_nodes[node_hash],
+                            terms))
+            for source in [self._get_tree(s, include_terms)
+                           for s in l.sources]:
+                if not (source, name) in self._edges:
+                    self._graph.attr('node', shape='ellipse',
+                                     style='transparent', color='black')
+                    self._graph.node(name)
+                    self._graph.edge(source, name)
+                    self._edges.add((source, name))
+        else:
+            terms = self._format_layer_terms(l) if include_terms else ''
+            name = ('{}'.format(self._format_description(l.description))
+                    if l.description else
+                    '{} ({}) {}'.format(l.type, self._sequence(), terms))
+            if l.type == 'FilterLayer':
+                self._graph.attr('node', shape='cds')
+                name += '  '
+            self._graph.node(name)
+            for ls in l.loss_sets:
+                ls_name = 'LossSet {}'.format(
+                    self._format_description(ls.description), self._sequence())
+                if not (ls_name, name) in self._edges:
+                    self._graph.attr('node',
+                                     shape='box',
+                                     style='filled',
+                                     color='lightgrey')
+                    self._graph.node(ls_name)
+                    self._graph.edge(ls_name, name)
+                    self._edges.add((ls_name, name))
+        return name
+
+    def _generate(self, lv_id, with_terms=True, fmt='pdf', rankdir='BT'):
+
+        # reset the internals for regeneration
+        self._sequence.reset()
+        self._unique_nodes.clear()
+        self._edges.clear()
+
+        # get the LayerView by id
+        lv = LayerView.retrieve(lv_id)
+
+        # build filename with format '<lv_id><-with_terms>-<rankdir>.<fmt>'
+        self._filename = lv.id
+        self._filename += '-with_terms' if with_terms else ''
+        self._filename += '-{}'.format(rankdir)
+
+        self._graph.graph_attr['rankdir'] = rankdir
+
+        # potentially update the format
+        self._graph.format = fmt
+
+        # now build the "tree" of nodes
+        self._get_tree(lv.layer, with_terms)
+
+    def view(self, lv_id, with_terms=True, fmt='pdf', rankdir='BT'):
```

Comment:
yes, I didn't really think of this as being used outside of ipython.  Not sure how useful it would be outside of a notebook.